File: frontend/src/types/submission.ts
export type ReadinessStatus = "ready" | "missing_info";
export type AdminStatus = "pending" | "approved" | "rejected";

export type Submission = {
  id: string;
  user_id: string;
  client_name: string;
  client_email: string;
  service_package: string;
  project_goal: string;
  desired_timeline: string;
  assets_provided: string;
  readiness_status: ReadinessStatus;
  missing_items: string[];
  ai_summary: string;
  recommended_next_action: string;
  admin_status: AdminStatus;
  asset_files: string[];
  asset_links: string[];
  brief_files: string[];
  brief_links: string[];
  brief_link: string;
  is_draft: boolean;
  created_at: string;
  updated_at: string;
};

export type CreateSubmissionInput = {
  client_name: string;
  client_email: string;
  service_package: string;
  project_goal: string;
  desired_timeline: string;
  assets_provided: string;
  // optional file fields handled via FormData, not stored in this type
};

export type CreateSubmissionResponse = {
  submission: Submission;
};

export type SubmissionsResponse = {
  submissions: Submission[];
};

export type UpdateSubmissionStatusResponse = {
  submission: Submission;
};

File: backend/internal/models/submission.go
package models

import "time"

const (
	ReadinessStatusReady       = "ready"
	ReadinessStatusMissingInfo = "missing_info"
	AdminStatusPending         = "pending"
	AdminStatusApproved        = "approved"
	AdminStatusRejected        = "rejected"
)

type Submission struct {
	ID                    string    `json:"id"`
	UserID                string    `json:"user_id"`
	ClientName            string    `json:"client_name"`
	ClientEmail           string    `json:"client_email"`
	ServicePackage        string    `json:"service_package"`
	ProjectGoal           string    `json:"project_goal"`
	DesiredTimeline       string    `json:"desired_timeline"`
	AssetsProvided        string    `json:"assets_provided"`
	ReadinessStatus       string    `json:"readiness_status"`
	MissingItems          []string  `json:"missing_items"`
	AISummary             string    `json:"ai_summary"`
	RecommendedNextAction string    `json:"recommended_next_action"`
	AdminStatus           string    `json:"admin_status"`
	AssetFiles            []string  `json:"asset_files"`
	AssetLinks            []string  `json:"asset_links"`
	BriefFiles            []string  `json:"brief_files"`
	BriefLinks            []string  `json:"brief_links"`
	BriefLink             string    `json:"brief_link"`
	IsDraft               bool      `json:"is_draft"`
	CreatedAt             time.Time `json:"created_at"`
	UpdatedAt             time.Time `json:"updated_at"`
}
